x/dex/keeper/core.go

```diff
@@ -331,49 +331,38 @@ func (k Keeper) PlaceLimitOrderCore(
 		return trancheKey, totalInCoin, swapInCoin, swapOutCoin, err
 	}
 
-	amountLeft, totalIn := amountIn, math.ZeroInt()
-
-	// For everything except just-in-time (JIT) orders try to execute as a swap first
-	if !orderType.IsJIT() {
-		// This is ok because tokenOut is provided to the constructor of PairID above
-		takerTradePairID := pairID.MustTradePairIDFromMaker(tokenOut)
-		var limitPrice math_utils.PrecDec
-		limitPrice, err = types.CalcPrice(tickIndexInToOut)
-		if err != nil {
-			return trancheKey, totalInCoin, swapInCoin, swapOutCoin, err
-		}
+	amountLeft := amountIn
+
+	// This is ok because tokenOut is provided to the constructor of PairID above
+	takerTradePairID := pairID.MustTradePairIDFromMaker(tokenOut)
+	var limitPrice math_utils.PrecDec
+	limitPrice, err = types.CalcPrice(tickIndexInToOut)
```

Comment:
a little out of scope, but how about renaming tickIndexInToOut -> limitTickIndexInToOut? if I understand it correctly,  it's acting as a tick limit, not as a tick of an actual swap